airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/TyperDeduperUtil.kt

```diff
@@ -0,0 +1,72 @@
+package io.airbyte.integrations.base.destination.typing_deduping
+
+import com.google.common.collect.Streams
+import io.airbyte.cdk.integrations.util.ConnectorExceptionUtil.getResultsOrLogAndThrowFirst
+import io.airbyte.commons.concurrency.CompletableFutures
+import java.util.*
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
+import java.util.concurrent.ExecutorService
+
+
+class TyperDeduperUtil {
```

Comment:
this class is partly taken from @jbfbell 's https://github.com/airbytehq/airbyte/pull/34637/files#diff-c9e7bf1074e965f63b135422dd2a7ac6ba196046b226a0ffe3a855a4eb82ed3b

it'll have some merge conflicts b/c I reordered the arguments in `prepareAllSchemas` but :shrug: just wanted to be consistent across the methods, and I generally prefer infra-ish params (executorservice, destinationhandler, etc.) to come before functional-ish params (parsedcatalog, streamconfig, etc.)